resources/tests/test_resource_api.py

```diff
@@ -136,11 +137,41 @@ def test_non_public_resource_visibility(api_client, resource_in_unit, user):
     response = api_client.get(url)
     assert response.status_code == 200
     assert response.data['count'] == 1
-
     url = reverse('resource-detail', kwargs={'pk': resource_in_unit.pk})
     response = api_client.get(url)
     assert response.status_code == 200
 
+    # Authenticated as unit admin
+    user.unit_authorizations.authorized = staff_user
+    user.unit_authorizations.level = UnitAuthorizationLevel.admin
+    user.unit_authorizations.subject = resource_in_unit.unit
+    user.save()
+    url = reverse('resource-list')
+    response = api_client.get(url)
+    assert response.status_code == 200
+    assert response.data['count'] == 1
+    assert Unit.objects.managed_by(user).values_list('id', flat=True)[0] == response.data['results'][0]['unit']
+
+    # Authenticated as unit group admin
+    user.unit_authorizations.level = UnitGroupAuthorizationLevel.admin
+    user.unit_authorizations.subject = resource_in_unit.unit
```

Comment:
I think you should clear the unit authorizations here and instead add a new unit *group* authorization.